crates/ty_python_semantic/src/types.rs

```diff
@@ -8294,6 +8330,56 @@ impl<'db> BoundSuperType<'db> {
     }
 }
 
+#[salsa::interned(debug)]
+pub struct TypeIsType<'db> {
+    return_type: Type<'db>,
+    /// The ID of the scope to which the place belongs
+    /// and the ID of the place itself within that scope.
+    place_info: Option<(ScopeId<'db>, ScopedPlaceId)>,
+}
+
+impl<'db> TypeIsType<'db> {
+    pub fn place_name(self, db: &'db dyn Db) -> Option<String> {
+        let (scope, place) = self.place_info(db)?;
+        let table = place_table(db, scope);
+        let expr = table.place_expr(place);
+
+        // TODO: Attribute and subscript
+        expr.as_name().map(ToString::to_string)
```

Comment:
This TODO is easy to address since `PlaceExpr` implements `Display`. But we can also leave it.